tests/phpunit/tests/db.php

```diff
@@ -1717,6 +1729,36 @@ public function data_prepare_with_placeholders() {
 				false,
 				"'hello' 'foo##'",
 			),
+			array(
+				'SELECT * FROM %i WHERE %i = %d;',
+				array( 'my_table', 'my_field', 321 ),
+				false,
+				'SELECT * FROM `my_table` WHERE `my_field` = 321;',
+			),
+			array(
+				'WHERE %i = %d;',
+				array( 'evil_`_field', 321 ),
+				false,
+				'WHERE `evil_``_field` = 321;', // To quote the identifier itself, then you need to double the character, e.g. `a``b`.
+			),
+			array(
+				'WHERE \'%i\' = 1 AND "%i" = 2 AND `%i` = 3 AND %15i = 4',
```

Comment:
What happens if someone does `AND ``%i`` `, out of curiousity? Anything that needs improving in cases like that?

Reply:
The Identifier would still be quoted, so you would get 3 back-ticks around it (I've added this to the test).

As to what will happen... MySQL/MariaDB will interpret the first ``` ` ``` as a start of an identifier, then the next 2 would be seen as a single back-tick as part of the identifier (doubling used to escape).

If you take this further, and start with 3 back-ticks, not only will it look wrong, you will be back to an SQL parse error (good)... and as you keep adding more, it alternates between the two states.